### ISSUE TITLE
Recursive model

### ISSUE THREAD

**Issue by Nicoc92** [JGCRI/gcam-core#479]
I have a question regarding how climate constraints, such as target temperature or radiative forcing for the year 2100, actually function within the GCAM model.
From what I understand, GCAM creates a CO2 market to limit emissions in order to meet these constraints. However, I’m not entirely sure I fully grasp how this mechanism works. Does the model first calculate emissions for 2100 and then iterate over each period to adjust the price of CO2 market (demand and supply) in order to meet the constraint set in the "target finder"?
I’m asking because I find it a bit difficult to understand how GCAM can impose this kind of constraint, given that it is a recursive model. Is there some optimization involved in this process? I would greatly appreciate more detailed information on how these constraints work and how they are enforced within the model.
Thank you in advance for your help,
Best regards,

**Comment by realxinzhao** [comment]
>  Does the model first calculate emissions for 2100 and then iterate over each period to adjust the price of the CO2 market (demand and supply) in order to meet the constraint set in the "target finder"

Yes, if `target-finding` through carbon prices. The assumption is carbon prices for emissions from fossil fuels grow at a Hotelling rate (3% annually).
